app/javascript/react/screens/App/Overview/components/Migrations/MigrationsCompletedList.js

```diff
@@ -1,205 +1,270 @@
+    isSortAscending: true
+  };
+
+  sortedMigrations = () => {
+    const { currentSortType, isSortNumeric, isSortAscending } = this.state;
+    const { finishedTransformationPlans } = this.props;
+
+    return sortFilter(currentSortType, isSortNumeric, isSortAscending, finishedTransformationPlans);
+  };
+
+  toggleCurrentSortDirection = () => {
+    this.setState(prevState => ({
+      isSortAscending: !prevState.isSortAscending
+    }));
+  };
+
+  updateCurrentSortType = sortType => {
+    const { currentSortType } = this.state;
+    if (currentSortType !== sortType) {
+      this.setState({
+        currentSortType: sortType,
+        isSortNumeric: sortType.isNumeric,
+        isSortAscending: true
+      });
+    }
+  };
+
+  render() {
+    const { sortFields, currentSortType, isSortNumeric, isSortAscending } = this.state;
+    const {
+      finishedTransformationPlans,
+      allRequestsWithTasks,
+      retryClick,
+      loading,
+      redirectTo,
+      showConfirmModalAction,
+      hideConfirmModalAction,
+      archiveTransformationPlanAction,
+      archiveTransformationPlanUrl,
+      fetchTransformationPlansAction,
+      fetchTransformationPlansUrl,
+      addNotificationAction,
+      archived
+    } = this.props;
+    const sortedMigrations = this.sortedMigrations();
 
-                  return fetchTransformationPlansAction({
-                    url: fetchTransformationPlansUrl,
-                    archived: false
+    return (
+      <Grid.Col xs={12}>
+        <Spinner loading={!!loading}>
+          {finishedTransformationPlans.length > 0 ? (
+            <React.Fragment>
+              {!archived && (
+                <React.Fragment>
+                  <Grid.Row>
+                    <Toolbar>
+                      <Sort>
+                        <Sort.TypeSelector
+                          sortTypes={sortFields}
+                          currentSortType={currentSortType}
+                          onSortTypeSelected={this.updateCurrentSortType}
+                        />
+                        <Sort.DirectionSelector
+                          isNumeric={isSortNumeric}
+                          isAscending={isSortAscending}
+                          onClick={this.toggleCurrentSortDirection}
+                        />
+                      </Sort>
+                    </Toolbar>
+                  </Grid.Row>
+                </React.Fragment>
+              )}
+              <ListView className="plans-complete-list" style={{ marginTop: 0 }}>
+                {sortedMigrations.map(plan => {
+                  const requestsOfAssociatedPlan = allRequestsWithTasks.filter(
+                    request => request.source_id === plan.id
+                  );
+
+                  const mostRecentRequest =
+                    requestsOfAssociatedPlan.length > 0 && getMostRecentRequest(requestsOfAssociatedPlan);
+
+                  const failed = mostRecentRequest && mostRecentRequest.status === 'Error';
```

Comment:
continuing w/ our theme here (and not related to this PR)... it would really be nice to pull these literals out into a constant `Error`, `Ok` etc and hoist them appropriately. I know that was introduced previously but just noting it. Do you mind if I put in a follow up PR for that later on?

Reply:
Sounds good!!